src/systems/game_stats_updater.rs

```diff
@@ -7,8 +7,9 @@ pub fn game_stats_updater_system(
 ) -> SystemResult {
     for ev in game_events.iter() {
         match ev {
-            GameEvent::DamageEntity(_, dmg) => game_stats.damage_dealt += dmg,
+            GameEvent::DamageEntity(_, _, dmg) => game_stats.damage_dealt += dmg,
             GameEvent::KillEntity(_) => game_stats.kill_count += 1,
+            GameEvent::TransferGold(_, _, gold) => game_stats.earned_gold += gold,
```

Comment:
This does not take into account gold that was gained from the gold gain multiplier

Reply:
oops, would you suggest to do the calculation of the multiplier again or pass the gold amount multiplied already?

Reply:
What about:
```
GameEvent::TransferGold(from, to)
GameEvent::TransferedGold(to, multiplied_quantity)
```

Reply:
Done!